### DISCUSSION TITLE
Built in UI Query Optimization

### DISCUSSION THREAD

**Discussion by FreeNFTj** [planetscale/discussion#380] [Feature Requests] [answered]
I'd be great to have an out of the box solution for query optimization. The insights is great - but plugins or even a house built solution to analyze if i'm using the correct indexes and how I can rewrite my query would be even better and save time. Something along the lines of a yellow warning on the query explorer with potential query rewrites based on my schema.

**Comment by orware** [comment, marked as answer]
Thank you for the suggestion @FreeNFTj! 

This type of solution is definitely on our minds and improvements to our Insights feature will continue to be made over time, and could include capabilities like what you've shared above in the future.

We don't have anything official to say on that topic at this time, but we wanted to make sure that you know that your suggestion has been heard and it is something that we are thinking about!

**Comment by amritk** [comment]
Yea currently as a front-end developer, I often leverage chat GPT to improve my sql queries. This can easily be done right on the query insights page. For bonus points you can even create a few branches from the AI suggestions on indexes/query optimization. Run the queries on said new branches, and find the optimal one.
